<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>

  <!-- Config and weights folder. -->
  <arg name="yolo_weights_path"          default="$(find darknet_ros)/yolo_network_config/weights"/>
  <arg name="yolo_config_path"           default="$(find darknet_ros)/yolo_network_config/cfg"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="darknet_ros" file="$(find object_tracking_yolo)/config/ros_op3.yaml"/>
  <rosparam command="load" ns="darknet_ros" file="$(find object_tracking_yolo)/config/yolov3.yaml"/>

  <!-- Start darknet and ros wrapper -->
  <node pkg="darknet_ros" type="darknet_ros" name="darknet_ros" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
  </node>
  
  <!-- UVC camera -->
  <!-- <node pkg="usb_cam" type="usb_cam_node" name="usb_cam_node" output="screen"> -->
    <!-- <param name="video_device" type="string" value="/dev/video0" /> -->
    <!-- <param name="image_width" type="int" value="1280" /> -->
    <!-- <param name="image_height" type="int" value="720" /> -->
    <!-- <param name="framerate " type="int" value="30" /> -->
    <!-- <param name="camera_frame_id" type="string" value="cam_link" /> -->
    <!-- <param name="camera_name" type="string" value="camera" /> -->
    <!-- <param name="autofocus" type="bool" value="False" /> -->
    <!-- <param name="autoexposure" type="bool" value="False" /> -->
    <!-- <param name="auto_white_balance" type="bool" value="False" /> -->
    <!-- <param name="gain" value="255" /> -->
    <!-- <param name="brightness" value="64" /> -->
    <!-- <param name="exposure" value="80" /> -->
    <!-- <param name="auto_exposure" type="bool" value="False" /> -->
    <!-- <param name="exposure_absolute" value="1000" /> -->
    <!-- <param name="auto_white_balance" type="bool" value="False" /> -->
    <!-- <param name="white_balance_temperature" value="2800" /> -->
    <!-- <param name="camera_info_url" type="string" value="file://$(find ar_pose)/data/camera_1280720.yaml" /> -->
  <!-- </node> -->

 <!--<node name="republish" type="republish" pkg="image_transport" output="screen" 	args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->
</launch>
